jax/lax/lax_parallel.py

```diff
@@ -48,6 +48,9 @@ def psum(x, axis_name):
     x: array(s) with a mapped axis named ``axis_name``.
     axis_name: hashable Python object used to name a pmapped axis (see the
       ``pmap`` docstring for more details).
+    axis_index_groups: optional list of lists containing axis indices (e.g. for
+      an axis of size 4, [[0, 1], [2, 3]] would perform psums over the first
+      two and last two replicas). All groups must be the same size.
```

Comment:
Maybe good to add that `axis_index_groups` must represent a partition of the integers {0, 1, ..., axis_size - 1} (i.e. it's a grouping of those integers into non-empty lists where each integer is included in a list exactly once) where the order of integers within each list matters.

Reply:
Added both a note and a check.